src/test/java/com/google/sps/TripServletTest.java

```diff
@@ -243,6 +246,46 @@ public void testPutTripDayInDatastore() throws Exception {
 
   @Test
   public void testPutEventsInDatastore() throws Exception {
+    // Mock the GeoApiContext object to be passed into PlacesApi methods,
+    // and the FindPlaceFromTextRequest.
+    GeoApiContext mockGeoApiContext = mock(GeoApiContext.class);
+    FindPlaceFromTextRequest findPlaceRequest = 
+      PowerMockito.mock(FindPlaceFromTextRequest.class);
+
+    // Create the FindPlaceFromText object with a valid place ID.
+    FindPlaceFromText findPlaceResult = new FindPlaceFromText();
+    findPlaceResult.candidates = new PlacesSearchResult[2];
+    findPlaceResult.candidates[0] = new PlacesSearchResult();
+    findPlaceResult.candidates[0].placeId = PLACE_ID_1;
+    findPlaceResult.candidates[1] = new PlacesSearchResult();
+    findPlaceResult.candidates[1].placeId = PLACE_ID_2;
+
+    // Have the findPlaceRequest.await() method return the FindPlaceFromText
+    // object using PowerMockito, as await() is a final method.
+    PowerMockito.when(findPlaceRequest.await()).thenReturn(findPlaceResult);
+
+    // Mock the PlacesApi object.
+    PowerMockito.mockStatic(PlacesApi.class);
+    when(PlacesApi.findPlaceFromText(any(), anyString(), any()))
+      .thenReturn(findPlaceRequest);
+
+    // Run the getPlaceIdFromTextSearch(...) method to test the result.
+    String placeIdResult = tripServlet.getPlaceIdFromTextSearch(mockGeoApiContext, 
+      TEXT_LOCATION_SEARCH);
+      
+    // set mock object behavior
+    when(request.getParameter("poi-1")).thenReturn(POI_ONE);
+    when(request.getParameter("poi-2")).thenReturn(POI_TWO);
+
+    // manually create params list
+    List<String> paramsList = new ArrayList<>();
+    paramsList.add("inputDestination");
+    paramsList.add("inputDayOfTravel");
+    paramsList.add("inputTripName");
+    paramsList.add("poi-1");
+    paramsList.add("poi-2");
+    Enumeration<String> params = Collections.enumeration(paramsList);
+
```

Comment:
Note: It'll likely be a good idea to put some of this into the Utility class.

Reply:
Agreed!